src/Chirp.Infrastructure/ChirpRepository/AuthorRepository.cs

```diff
@@ -7,130 +7,161 @@
 
 namespace Chirp.Infrastructure.ChirpRepository;
 
-public class AuthorRepository : IAuthorRepository<Author, Cheep>
+public class AuthorRepository : IAuthorRepository<Author, Cheep, User>
 {
-    protected DbSet<Author> DbSet;
+    protected DbSet<Author> DbSetAuthor;
+    protected DbSet<User> DbSetUser;
     protected ChirpDBContext context;
 
     public AuthorRepository(ChirpDBContext dbContext)
     {
-        DbSet = dbContext.Authors;
+        DbSetAuthor = dbContext.Authors;
         context = dbContext;
     }
 
     #region IAuthorRepository<Author, Cheep> Members
 
     public void Insert(Author entity)
     {
-        DbSet.Add(entity);
+        DbSetAuthor.Add(entity);
         context.SaveChanges();
     }
 
     public void Delete(Author entity)
     {
-        DbSet.Remove(entity);
+        DbSetAuthor.Remove(entity);
+        context.SaveChanges();
     }
 
     public IQueryable<Author> SearchFor(Expression<Func<Author, bool>> predicate)
     {
-        return DbSet.Where(predicate);
+        return DbSetAuthor.Where(predicate);
     }
 
     public async Task<Author> GetAuthorWithCheeps(string authorName)
     {
         // The returned Author object can use Author.Cheeps to get all the Cheeps (sorted by descending timestamp)
         
-        var author = await DbSet.Include(_author => _author.Cheeps
+        var author = await DbSetAuthor.Include(_author => _author.Cheeps
                     .OrderByDescending(_cheep => _cheep.TimeStamp))
-                    .Where(_author => _author.Name == authorName)
+                    .Where(_author => _author.User.Name == authorName)
                     .FirstOrDefaultAsync() ?? throw new Exception($"Author {authorName} not found");
         return author;
     }
 
+    public async Task<int> GetCheepsCountsFromAuthorId(int id) 
+    {
+        var authorEntity = SearchFor(_author => _author.User.UserId == id).FirstOrDefault();
+
+        if (authorEntity == null) {
+            return 0; // make this into an exception
+        }
+
+        int cheepsCount = DbSetAuthor.Entry(authorEntity)
```

Comment:
If there's a `CountAsync`, I'd recommend using that and adding an await operator as such: `int cheepsCount = await DbSetAuthor [...]`

Reply:
There is indeed a countasync and it has been changed to that